src/java/com/zimbra/cs/nginx/NginxLookupExtension.java

```diff
@@ -880,13 +880,23 @@ private DomainExternalRouteInfo getDomainExternalRouteInfo(ILdapContext zlc, Con
 
         private Server lookupUpstreamImapServer(NginxLookupRequest req) throws ServiceException {
             ImapLoadBalancingMechanism LBMech = ImapLoadBalancingMechanism.newInstance();
-            Server server = prov.getLocalServer();
-            String[] imapServerAddrs;
-            if (server != null) {
-                imapServerAddrs = server.getReverseProxyUpstreamImapServers();
-            } else {
-                imapServerAddrs = prov.getConfig().getReverseProxyUpstreamImapServers();
+            String[] imapServerAddrs = null;
+            Account acct = prov.get(AccountBy.name, req.user);
+            if (acct != null) {
+                imapServerAddrs = acct.getServer().getReverseProxyUpstreamImapServers();
```

Comment:
yes. Account's record may have empty value in zimbraMailHost. The value is not required to be non-empty until account's mailbox is created.